src/internal/m365/collection/drive/collection.go

```diff
@@ -72,13 +72,14 @@ type Collection struct {
 	// Specifies if it new, moved/rename or deleted
 	state data.CollectionState
 
-	// scope specifies what scope the items in a collection belongs
-	// to. This is primarily useful when dealing with a "package",
-	// like in the case of a OneNote file. A OneNote file is a
-	// collection with a package scope and multiple files in it. Most
-	// other collections have a scope of folder to indicate that the
-	// files within them belong to a folder.
-	scope collectionScope
+	// true if this collection, or a parent directory of this collection,
+	// is marked as a package.
+	// packages are only marked on the top-level directory, but special-case
+	// handling need apply to all subfolders.  Therefore it is necessary to cascade
+	// that identification to all affected collections, not just those that identify
+	// as packages themselves.
+	// see: https://learn.microsoft.com/en-us/graph/api/resources/package?view=graph-rest-1.0
+	isPackageOrChildOfPackage bool
```

Comment:
How about just `isPackage`?

Reply:
going to keep for now, since it clearly idicates the conditions for the flag.  two-way door; so if we hate it in the future, we can rename as needed.